File: 11_ContainerWithMostWater/python/ContainerWithMostWater.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0
        right = len(height)-1
        size = 0
        while left<right:
            flg = 0
            m_size = min(height[left], height[right]) * (right - left)
            if m_size>size:
                size = m_size
            if height[left]<height[right]:
                for i in range(left+1, right):
                    if height[i]>height[left]:
                        left = i
                        flg = 1
                        break
            else:
                for i in range(right-1, left,-1):
                    if height[i]>height[right]:
                        right = i
                        flg = 1
                        break
            if flg == 0:
                break
        return size

File: 11_ContainerWithMostWater/c#/ContainerWithMostWater.cs
public class Solution {
    public int MaxArea(int[] height) {
        int left = 0;
        int right = height.Length-1;
        int res_size = 0;
        bool flg = true;
        while(left<right){
            flg = true;
            // cout<<left<<":"<<right<<endl;
            if(height[left]>height[right]){
                if(height[right]*(right-left)>res_size){
                    res_size = height[right]*(right-left);
                }
                for(int i = right-1; i>left; i--){
                    if(height[i]>height[right]){
                        flg = false;
                        right = i;
                        break;
                    }
                }
                
            }
            
            
            else{
                if(height[left]*(right-left)>res_size){
                    res_size = height[left]*(right-left);
                }
                for(int i = left+1; i<right; i++){
                    if(height[i]>height[left]){
                        flg = false;
                        left = i;
                        break;
                    }
                }
                
            }
            
            if(flg){
                break;
            }
        }
        
        return res_size;
    }
}